src/patternfly/components/DataToolbar/data-toolbar.scss

```diff
@@ -0,0 +1,351 @@
+  --pf-c-data-toolbar__group--m-toggle-group--m-reveal--spacer: var(--pf-c-data-toolbar--spacer--base);
+  --pf-c-data-toolbar__toggle--c-badge--MarginLeft: var(--pf-global--spacer--sm);
+
+  // Button group
+  --pf-c-data-toolbar__group--m-button-group--spacer: var(--pf-c-data-toolbar--spacer--base);
+  --pf-c-data-toolbar__group--m-button-group--space-items: var(--pf-global--spacer--sm);
+
+  // Icon button group spacer
+  --pf-c-data-toolbar__group--m-icon-button-group--spacer: var(--pf-c-data-toolbar--spacer--base);
+  --pf-c-data-toolbar__group--m-icon-button-group--space-items: 0;
+
+  // Icon group spacer
+  --pf-c-data-toolbar__group--m-filter-group--spacer: var(--pf-c-data-toolbar--spacer--base);
+  --pf-c-data-toolbar__group--m-filter-group--space-items: 0;
+
+  // Condensed spacer
+  --pf-c-data-toolbar__group--m-condensed--space-items: 0;
+
+  // Bulk select
+  --pf-c-data-toolbar--m-bulk-select--spacer: var(--pf-global--spacer--lg);
+
+  // Overflow menu item
+  --pf-c-data-toolbar--m-overflow-menu--spacer: var(--pf-c-data-toolbar--spacer--base);
+
+  // Label
+  --pf-c-data-toolbar--m-label--spacer: var(--pf-global--spacer--md);
+  --pf-c-data-toolbar--m-label--FontWeight: var(--pf-global--FontWeight--bold);
+
+  // Search filter
+  --pf-c-data-toolbar--m-search-filter--spacer: var(--pf-global--spacer--sm);
+
+  // Separator
+  --pf-c-data-toolbar--m-separator--spacer: var(--pf-c-data-toolbar--spacer--base);
+  --pf-c-data-toolbar--m-separator--BackgroundColor: var(--pf-global--BorderColor--100);
+  --pf-c-data-toolbar--m-separator--Width: var(--pf-global--BorderWidth--md);
+  --pf-c-data-toolbar--m-separator--Height: var(--pf-global--FontSize--lg);
+
+  position: relative;
+  background-color: var(--pf-c-data-toolbar--BackgroundColor);
+
+  // Separator
+  .pf-m-separator {
+    --pf-c-data-toolbar--spacer: var(--pf-c-data-toolbar--m-separator--spacer);
+
+    align-self: center;
+    width: var(--pf-c-data-toolbar--m-separator--Width);
+    height: var(--pf-c-data-toolbar--m-separator--Height);
+    background-color: var(--pf-c-data-toolbar--m-separator--BackgroundColor);
+  }
+
+  // Group
+  .pf-m-button-group {
+    --pf-c-data-toolbar--spacer: var(--pf-c-data-toolbar__group--m-button-group--spacer);
+
+    > * {
+      --pf-c-data-toolbar--spacer: var(--pf-c-data-toolbar__group--m-button-group--space-items);
+    }
+  }
+
+  // Icon button group
+  .pf-m-icon-button-group {
+    --pf-c-data-toolbar--spacer: var(--pf-c-data-toolbar__group--m-icon-button-group--spacer);
+
+    > * {
+      --pf-c-data-toolbar--spacer: var(--pf-c-data-toolbar__group--m-icon-button-group--space-items);
+    }
+  }
+
+  // Filter group
+  .pf-m-filter-group {
+    --pf-c-data-toolbar--spacer: var(--pf-c-data-toolbar__group--m-filter-group--spacer);
+
+    > * {
+      --pf-c-data-toolbar--spacer: var(--pf-c-data-toolbar__group--m-filter-group--space-items);
+    }
+
+    > * + * {
+      margin-left: -1px;
```

Comment:
sweeet. The only issue with this (and in the input group) is if you focus on an element that's to the left of another element in the group, the focus ring is hidden behind the right element, since the right element overlaps the left and has a higher stacking order since it comes after in the DOM.

Maybe we could use `:focus-within`? Here's a codepen as an example - https://codepen.io/mcoker/pen/BaBKORg. I tried it in the toolbar and needed to add a `z-index`, too, but didn't look at it much further as to why. Not sure that we want to add a `z-index`, which could cause random issues with that element popping out from underneath other elements.

Reply:
`:focus-within` didn't work well for `.pf-c-input-group` text input. Setting `:focus` on `__item` children worked. My main concern w/setting `z-index` was related to dropdowns within the same toolbar. Setting the `z-index` value to `var(--pf-global--ZIndex--xs)` ensures that a `:focus`ed toolbar element won't appear above a dropdown. Other `z-index`ed elements have higher `z-index`es, but it could still see it causing issues..